sdk/python/feast/pyspark/batch_retrieval_job.py

```diff
@@ -0,0 +1,378 @@
+    query_conf: List[Dict[str, Any]], entity: DataFrame, tables: Dict[str, DataFrame]
+) -> DataFrame:
+    """Perform as of join between entity and multiple feature table. Returns a DataFrame.
+
+    Args:
+        query_conf (List[Dict[str, Any]]):
+            Query configuration.
+        entity (DataFrame):
+            Entity dataframe. Must contain the column event_timestamp.
+        tables (Dict[str, DataFrame]):
+            Map of feature table name to Spark DataFrame.
+
+    Returns:
+        DataFrame: Join result.
+
+    Example:
+        >>> entity.show()
+            +------+-------------------+
+            |entity|    event_timestamp|
+            +------+-------------------+
+            |  1001|2020-09-02 00:00:00|
+            +------+-------------------+
+
+        >>> feature1.show()
+            +------+--------+-------------------+-------------------+
+            |entity|feature1|    event_timestamp|  created_timestamp|
+            +------+--------+-------------------+-------------------+
+            |    10|     200|2020-09-01 00:00:00|2020-09-01 00:00:00|
+            +------+--------+-------------------+-------------------
+
+        >>> feature2.show()
+            +------+--------+-------------------+-------------------+
+            |entity|feature2|    event_timestamp|  created_timestamp|
+            +------+--------+-------------------+-------------------+
+            |    10|     400|2020-09-01 00:00:00|2020-09-01 00:00:00|
+            +------+--------+-------------------+-------------------
+
+
+        >>> tables = {"table1": feature1, "table2": feature2}
+
+        >>> query_conf = [
+                {
+                    "table": "table1",
+                    "features": ["feature1"],
+                    "join": ["entity"],
+                },
+                {
+                    "table": "table2",
+                    "features": ["feature2"],
+                    "join": ["entity"],
+                },
+            ]
+
+        >>> joined_df = join_entity_to_feature_tables(
+                query_conf,
+                entity,
+                tables
+            )
+
+        >>> joined_df.show()
+            +------+-------------------+----------------+----------------+
+            |entity|    event_timestamp|table1__feature1|table2__feature2|
+            +------+-------------------+----------------+----------------+
+            |  1001|2020-09-02 00:00:00|             200|             400|
+            +------+-------------------+----------------+----------------+
+    """
+    joined = entity
+    for query in query_conf:
+        joined = as_of_join(
+            joined,
+            query["join"],
+            tables[query["table"]],
+            query["features"],
+            feature_prefix=f"{query['table']}__",
+            max_age=query.get("max_age"),
+        )
+    return joined
+
+
+def batch_retrieval(spark: SparkSession, conf: Dict) -> DataFrame:
```

Comment:
Should we use a verb in this method `retrieve_historical_features`? We have moved on from the term `batch`